modin/core/storage_formats/base/query_compiler.py

```diff
@@ -2762,6 +2762,7 @@ def groupby_agg(
         agg_kwargs,
         how="axis_wise",
         drop=False,
+        self_is_series=False,
```

Comment:
we can probably use the query compiler's `shape_hint` field instead of this parameter

example:
https://github.com/modin-project/modin/blob/c2b8d8e20b37a6ea6d3111112181984e51f3208b/modin/pandas/dataframe.py#L2582-L2588

Reply:
While this is definitely possible, it requires changing `._shape_hint` from being just a _hint_ into a proper flag which is always maintained correctly. For instance, in the example you're showing above it is IMHO wrong to put `column` there without a check - what if it is actually a `row`? We aren't checking if it is `1xN`- or `Nx1`-shaped in this particular place.

I'll add a `FIXME`.

Reply:
Well, I had to add it back, as `.shape_hint` at query compiler level does not behave the way I expected - it basically it behaving like "is this 1xN or Nx1" regardless of whether it is Series or DataFrame. So, when we do a `df[['foo']]`, we (correctly) get a 1xN DataFrame, but its hint is set to `column`.

And I need a way to tell them apart at groupby because pandas behave differently in some cases...